packages/ethereum-storage/src/ipfs-manager.ts

```diff
@@ -109,16 +109,14 @@ export default class IpfsManager {
         method: 'post',
         path: this.IPFS_API_ADD,
         port: this.ipfsConnection.port,
+        timeout: this.ipfsConnection.timeout > 0 ? this.ipfsConnection.timeout : undefined,
       });
+
       // Sending the request
       addForm.pipe(addRequest);
 
-      if (this.ipfsConnection.timeout && this.ipfsConnection.timeout > 0) {
-        addRequest.setTimeout(this.ipfsConnection.timeout);
-      }
```

Comment:
This is the fix. The timeout can only be set once the socket is open. So 2 solutions:
- either listen for the socket "open" event, and then set the timeout
- or set it directly via the new "timeout" option, directly at request creation (what I did here)